dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/features/InfrastructureApi.java

```diff
@@ -76,4 +85,63 @@
    @Fallback(Fallbacks.EmptyPagedIterableOnNotFoundOr404.class)
    PagedIterable<OperatingSystem> listOperatingSystems(@QueryParam("datacenterId") String datacenterId);
 
+   final class ParseDatacenters extends ParseJson<Datacenters> {
+
+      @Inject
+      ParseDatacenters(Json json) {
+         super(json, TypeLiteral.get(Datacenters.class));
+      }
+
+      private static class ToPagedIterable extends Arg0ToPagedIterable<Datacenter, ToPagedIterable> {
+
+         private DimensionDataCloudControlApi api;
+
+         @Inject
+         ToPagedIterable(DimensionDataCloudControlApi api) {
+            this.api = api;
+         }
+
+         @Override
+         protected Function<Object, IterableWithMarker<Datacenter>> markerToNextForArg0(Optional<Object> arg) {
+            return new Function<Object, IterableWithMarker<Datacenter>>() {
+               @Override
+               public IterableWithMarker<Datacenter> apply(Object input) {
+                  PaginationOptions paginationOptions = PaginationOptions.class.cast(input);
+                  return api.getInfrastructureApi().listDatacenters(paginationOptions);
+               }
+            };
+         }
+      }
+   }
+
+   final class ParseOperatingSystems extends ParseJson<OperatingSystems> {
+
+      @Inject
+      ParseOperatingSystems(Json json) {
+         super(json, TypeLiteral.get(OperatingSystems.class));
+      }
+
+      private static class ToPagedIterable extends Arg0ToPagedIterable<OperatingSystem, ToPagedIterable> {
+
+         private DimensionDataCloudControlApi api;
+
+         @Inject
+         ToPagedIterable(DimensionDataCloudControlApi api) {
+            this.api = api;
+         }
+
+         @Override
+         protected Function<Object, IterableWithMarker<OperatingSystem>> markerToNextForArg0(
+               final Optional<Object> arg) {
+            return new Function<Object, IterableWithMarker<OperatingSystem>>() {
+               @Override
+               public IterableWithMarker<OperatingSystem> apply(Object input) {
+                  PaginationOptions paginationOptions = PaginationOptions.class.cast(input);
+                  return api.getInfrastructureApi().listOperatingSystems(arg.toString(), paginationOptions);
```

Comment:
This should have been spotted by the Mock test. Let's modify the mock tests that use pagination and verify *all* the requests being sent. The second one should contain the pagination parameters and we should also verify that request.